packages/gafl-webapp-service/src/schema/validators/validators.js

```diff
@@ -26,18 +33,33 @@ export const startDateValidator = payload => {
     Joi.object({ 'licence-to-start': Joi.string().valid(AFTER_PAYMENT, ANOTHER_DATE).required() })
   )
   if (payload[LICENCE_TO_START_FIELD] === ANOTHER_DATE) {
-    const minDate = moment().tz(SERVICE_LOCAL_TIME).startOf('day')
-    const maxDate = moment().tz(SERVICE_LOCAL_TIME).add(ADVANCED_PURCHASE_MAX_DAYS, 'days')
-
     const day = payload['licence-start-date-day']
     const month = payload['licence-start-date-month']
     const year = payload['licence-start-date-year']
 
-    Joi.assert(dateSchemaInput(day, month, year), dateSchema)
-    const startDate = moment(`${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`, 'YYYY-MM-DD')
-      .tz(SERVICE_LOCAL_TIME)
-      .startOf('day')
-      .toDate()
-    Joi.assert({ startDate }, Joi.object({ startDate: Joi.date().min(minDate.toDate()).max(maxDate.toDate()) }))
+    const minDate = moment().tz(SERVICE_LOCAL_TIME).startOf('day').toDate()
+    const maxDate = moment().tz(SERVICE_LOCAL_TIME).add(ADVANCED_PURCHASE_MAX_DAYS, 'days').toDate()
+    validateDate(day, month, year, minDate, maxDate)
   }
 }
+
+export const getErrorFlags = error => {
```

Comment:
Suggest naming this `getDateErrorFlags` just so the specific purpose is more clear.